### PR TITLE
Update dependency constraints, fix deprecation warnings

### PR DESCRIPTION
**Pull request by Viicos**
## Overview: What does this pull request change?

Mainly:
- Loosen library dependencies (WIP)
- Switch from the deprecated `dev-dependencies` section to Poetry groups
- Remove the use of `requests` in the command handling. A bit of cleaning is also made there
- Drop unused dependencies
- Excplicity state we are using a vendored version of `click-default-group`
- Fix some deprecation warnings
- Update almost all dev dependencies

## Motivation and Explanation: Why and how do your changes improve the library?

Even if most people will install Manim in an isolated environment (e.g. a _single_ virtualenv for Manim), we still have to consider Manim as an installable package, meaning it can live with other unrelated libraries in the same environment (one example would be when using plugins).

Currently, Manim is using [caret requirements](https://python-poetry.org/docs/dependency-specification/#caret-requirements) (`^`), implicitly setting upper bounds for dependencies. Most people consider this as a bad practice, for two main reasons:
- each time a new "major" version is out, we would have to update our dependencies (and looking at the current state of the `pyproject.toml`, a couple ones aren't).
- If this isn't handled, users of manim are bound to versions they don't control. Say I want to use a feature in `watchdog>=3.0.0` inside my virtualenv where Manim is installed, I'd have to open a PR on the Manim repo to loosen the dependency on `watchdog` (which currently is `>=2.1.6,<3.0.0`), and there're probably no breaking changes for this version bump. 

My goal here is to go over the dependencies one by one, and set a reliable minimum version (to be sure it isn't incompatible with how it is used in the codebase).

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -27,26 +27,25 @@ packages = [
...
 rich = ">=6.0,!=12.0.0"
-pycairo = "^1.21"
-manimpango = "^0.4.0.post0"
-networkx = "^2.5"
-decorator = "^5.0.7"
+pycairo = ">=1.13"
+manimpango = "<=1.0.0"  # Complete API change in 1.0.0
+networkx = ">=2.6"
+decorator = ">=4.3.2"
 importlib-metadata = { version = "^4.10.0", python = "<3.8" }
```
Comment (Viicos) [inline pyproject.toml:42]: Is it ever used in Manim? Can't find any references to it

Comment (Viicos) [reply pyproject.toml]: Removed.

**Comment by Viicos** [comment]
> This is a good idea in general, and the tests seem like they *almost* pass. I mainly wanted to get rid of all the individual (automated) PRs by collecting them together here. 
>
> I haven't had enough time to carefully check the modified dependencies on your PR yet -- but there are some core dependencies where I'd likely want to stick with caret restrictions (e.g., numpy, pycairo, moderngl, manimpango ...) because the risk of breaking changes is too high. I'll leave a comment on #3376 later today.

_Originally posted by @behackl in https://github.com/ManimCommunity/manim/issues/3399#issuecomment-1755174236_

I've added constraints to these specific libraries: `pycairo`, `moderngl`, `manimpango`. Regarding `numpy` and `scipy`, I don't think adding an upper constraint is necessary: their API is more than stable, and they both emit deprecation warnings in advance. I'm going to check for deprecation warnings when running tests, and check if they pass with numpy 2.0.0 (from master) as well.

File: pyproject.toml [hunk 2]
```diff
@@ -27,86 +27,71 @@ packages = [
...
-tqdm = "^4.62.3"
-pydub = "^0.25.1"
+python = ">=3.9,<3.13"
+click = ">=8.0"
+cloup = ">=2.0.0"
+dearpygui = { version = ">=1.0.0", optional = true }
+decorator = ">=4.3.2"
+isosurfaces = ">=0.1.0"
+jupyterlab = { version = ">=3.0.0", optional = true }
+manimpango = "<=1.0.0"  # Complete API change in 1.0.0
```
Comment (naveen521kk) [inline pyproject.toml:37]: I would suggest adding a lower bound here, just like how it previously was. Also, I think shouldn't include `=` in the version, it should be `<1.0.0`.

**Comment by Viicos** [comment]
@behackl This is finally ready for review and tests passes. The diff may seem overloaded but most of the edited files are imports cleanup (from `click` to `cloup`).

**Review by behackl — APPROVED** [review]
Seems fine to me. Thank you very much for your efforts!

**Pull request merged**